### PR TITLE
api: add vault delete

### PR DESCRIPTION
**Pull request by dcow**

### CODE REVIEW

**Comment by dcow** [comment]
@tremblay here's vault deleted

File: api/src/lib.rs [hunk 1]
```diff
@@ -328,6 +328,7 @@ where
             };
             let vb_new = serde_json::to_vec(&vault).map_err(server_err)?;
             db.put(&id, &vb_new).await.map_err(server_err)?;
+            db.del_version("v1", id).await.map_err(server_err)?;
```
Comment (tremblay) [inline api/src/lib.rs:331]: this `v1` isn't related to our API version, right?

**Review by tremblay — DISMISSED** [review]
I don't know rust at all, but it seems reasonable...

File: api/src/lib.rs (hunk 1 above)
Comment (dcow) [reply api/src/lib.rs:331]: this removes the old vault data once it's been migrated. it's kinda related. why do you ask?

Comment (tremblay) [reply api/src/lib.rs:331]: That makes sense, I was just worried that it could mean the delete action wouldn't work in v2

**Review by ceberly — DISMISSED** [review]
Looks good, we share the code to turn a [u8] into a public key in a lot of requests, we could move it into a helper function or middleware...

**Review by ceberly — APPROVED** [review]
ok!

**Pull request merged**